ksqldb-common/src/main/java/io/confluent/ksql/util/timestamp/PartialStringToTimestampParser.java

```diff
@@ -51,11 +51,10 @@ public long parse(final String text) {
 
     if (text.contains("T")) {
       date = text.substring(0, text.indexOf('T'));
-      final String withTimezone = completeTime(
-          text.substring(text.indexOf('T') + 1)
-      );
+      final String withTimezone = text.substring(text.indexOf('T') + 1);
       timezone = getTimezone(withTimezone);
-      time = completeTime(withTimezone.substring(0, withTimezone.length() - timezone.length()));
+      time = completeTime(withTimezone.substring(0, withTimezone.length() - timezone.length())
+              .replace("Z",""));
```

Comment:
This adds the syntax that we want to support, but it also allows for additional non-ISO-8601 strings to be parsed into dates.

For instance, I think you could add as many trailing ZZZs as you like to a timestamp and it would be parsed.

Reply:
@jnh5y We have these:

- https://docs.ksqldb.io/en/latest/reference/sql/time/ (this is probably where we'd want to insert a note)
- https://docs.ksqldb.io/en/latest/reference/sql/data-types/#time-types
- https://docs.ksqldb.io/en/latest/concepts/time-and-windows-in-ksqldb-queries/